src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonConverterOfT.cs

```diff
@@ -332,7 +331,7 @@ internal bool TryWrite(Utf8JsonWriter writer, in T value, JsonSerializerOptions
             bool ignoreCyclesPopReference = false;
 
             if (
-#if NET6_0_OR_GREATER
+#if NET5_0_OR_GREATER
```

Comment:
Why do we #ifdef here? Can we just use the cached `IsValueType` value always?

Reply:
Recent versions of the JIT (5.0 or later) treat `typeof(T).IsValueType` as a constant and can therefore optimize away branching. Cf. #1157